src/main.js

```diff
@@ -322,14 +322,36 @@ class MiniGraphCard extends LitElement {
     `;
   }
 
+  renderLoader() {
+    const { height } = this.config;
+    const numberOfBars = 20;
+    return svg`
+        <div style="align-items: center; justify-content: center; display: flex; transform: rotate(180deg);">
+            <svg width="100%" height="100%" viewBox="0 0 500 ${height}">
+                ${Array.from({ length: numberOfBars }, (_, i) => i).map(i => svg`
+                    <rect x="${(i * (numberOfBars + 3)) + 25}" y="10" width="10" height="${this.getLoaderBarHeight(height, i)}" fill="var(--secondary-text-color)">
+                        <animate attributeName="opacity" calcMode="spline" dur="2s" values="0;1;0" keyTimes="0;0.5;1" keySplines="0.42,0,0.58,1;0.42,0,0.58,1" repeatCount="indefinite"/>
+                    </rect>
+                `)}
+            </svg>
+        </div>
+    `;
+  }
+
+  getLoaderBarHeight(height, i) {
+    return (Math.abs(Math.sin(i / 2 + 1) * height / 1.5) % height) + 10;
+  }
+
   renderGraph() {
+    const ready = this.entity[0] && this.Graph[0]._history !== undefined;
+
     return this.config.show.graph ? html`
       <div class="graph">
         <div class="graph__container">
           ${this.renderLabels()}
           ${this.renderLabelsSecondary()}
           <div class="graph__container__svg">
-            ${this.renderSvg()}
+            ${ready ? this.renderSvg() : this.renderLoader()}
```

Comment:
I would agree that using a standard progress indicator is better...
@JulienDeveaux please try to use that standard component.

P.S. How can I test this?
What I can do it just pressing F5, then a "pulsating graph" is visible for a blink.

Reply:
I tried replacing the svg I did with the ha-circular-progress loader, but it doesn't show anything on my end.
Did you test it ?

Reply:
This `ha-circular-progress` is a component derived from a "material/web/progress/circular-progress" element - check "src/components/ha-circular-progress.ts" file (HA frontend).
Just a guess: you should probably import that `MdCircularProgress` class and define `--md-sys-color-primary` & `--md-circular-progress-size` like it is done in that frontend file.

Reply:
Would that be duplication ?
Do you know a way to reuse the ha-circular-progress from our side ?
I don't think I can import it :/

Reply:
This worked for me: 
```js
${ready ? this.renderSvg() : html`<ha-circular-progress indeterminate></ha-circular-progress>`}
```

I did set the CSS manually in the Chrome inspector, though.

Reply:
> I did set the CSS manually in the Chrome inspector, though

Then create me a test js file - I will tell you then what to fix to locate it properly.

Reply:
Don't worry, I made it work (I was wrapping it in svg`` not html`` ._.)
I'll test it in different cases and it should be good

Reply:
Done

Reply:
Can you tell me how can I simulate a "loading" condition for testing?

Reply:
I simulate an infinite loading like so : ${ready ? this.renderSvg() : this.renderLoader()} -> ${ready ? this.renderSvg() : this.renderSvg()}

Reply:
@JulienDeveaux Can you show the indicator permanently for testing?

Reply:
Yeah, just remove renderSvg and replace it with renderLoader. You will get an infinite loader -> ${ready ? this.renderSvg() : this.renderSvg()}

Reply:
Erm, how shall I do it in an obfuscated js file? )) I am testing with a ready file, not compiling the project into a js file...

Reply:
> Done

Thanks! I think, there's still some small streamlining possible:
* We should replace the entire `graph__container` by the `<ha-circular-progress>`, because also labels are not useful while loading.
* I think, the only required CSS on the loader is `margin: auto`. We should set it in the actual CSS part of the code.
* Maybe we then want to avoid creating the new renderLoader function. But it might also be handy to have, in case we need to add extra logic later.

Reply:
Oh right 🤔
Let me send you a compiled file

Reply:
@JulienDeveaux I think it is already not necessary!
Managed to test

Reply:
Here is the look:
![image](https://github.com/user-attachments/assets/af8cc301-8155-4de2-b00b-f154e09aa00b)
The indicator located in the center. Imho there is nothing to fix here for css.

Reply:
You could also mangle it here and the CI will do the compiling for each change we discuss. We just need to remember to fix it again before committing. We could open a review comment there to not forget... Or set the PR to draft state.

Reply:
> We should replace the entire graph__container by the <ha-circular-progress>, because also labels are not useful while loading.

Agree.

Reply:
I've moved it and remove the extra class it was indeed unnecessary